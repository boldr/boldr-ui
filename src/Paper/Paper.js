/* eslint-disable react/default-props-match-prop-types */
// @flow
import React from 'react';
import type { Node } from 'react';
import cn from 'classnames';

type Props = {
  //  The component to render the paper as.
  component: Function | string,
  //  An optional css className to apply.
  className: ?string,
  // The depth of the paper. This should be a number between 0 - 5. If
  // the depth is 0, it will raise to a depth of 3 on hover.
  zDepth: number,
  // Any children to display in the paper.
  children: Array<Node>,
  //  Boolean if the paper should raise to the `zDepth` of `3`
  //  on hover when the initial
  raiseOnHover: boolean,
  // whether we apply a border raidus or not.
  isRounded: boolean,
  isPadded?: boolean,
};

const Paper = (props: Props) => {
  const {
    component: Component,
    zDepth,
    className,
    raiseOnHover,
    isRounded,
    isPadded,
    ...rest
  } = props;

  return (
    <Component
      {...rest}
      className={cn(
        `boldr-paper boldr-paper__${zDepth}`,
        {
          'boldr-paper__0--hover': zDepth === 0 && raiseOnHover,
          'boldr-paper--round': isRounded,
          'boldr-paper--no-pad': !isPadded,
        },
        className,
      )}
    />
  );
};

const defaultProps = {
  zDepth: 1,
  component: 'div',
  isRounded: false,
  isPadded: false,
};

Paper.defaultProps = defaultProps;

export default Paper;
