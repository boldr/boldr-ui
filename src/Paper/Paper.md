# Paper

The `Paper` component is a simple wrappper that adds box-shadow.

You can also use the scss mixin instead of paper.

### Usage

```scss
  @include md-shadow(5);
```

```javascript
  <Paper zDepth={ 2 } raiseOnHover>
    child elements
  </Paper>
```

### Props:

Name    Type  Definition
--- | --- | ----
component | Function / string | The component to render the paper as. Default div
zDepth | number |The depth of the paper. This should be a number between 0 - 5.
raiseOnHover  | boolean | Boolean if the paper should raise to the `zDepth` of `3` on hover when the initial
children | ReactChildren | Any children to display in the paper.
className | string | custom css class name
