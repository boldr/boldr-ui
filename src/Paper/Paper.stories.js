import React from 'react';
import { storiesOf } from '@storybook/react';

import Paper from './Paper';

storiesOf('Paper', module)
  .add('default', () => (
    <Paper>
      Wolf whatever craft beer, readymade blue bottle raclette squid next level af franzen heirloom
      90's XOXO beard. Twee schlitz health goth cray. Biodiesel butcher coloring book venmo
      snackwave vexillologist, neutra live-edge enamel pin chia. Roof party church-key fanny pack
      sartorial, flannel shabby chic four loko literally tumeric stumptown coloring book. Organic
      scenester mumblecore cliche, vice readymade helvetica. Glossier 8-bit disrupt affogato pop-up
      post-ironic, la croix cardigan. Gentrify tousled scenester, iceland dreamcatcher master
      cleanse kombucha hoodie mlkshk cardigan seitan before they sold out drinking vinegar ethical
      kickstarter.
    </Paper>
  ))
  .add('without padding', () => (
    <Paper isPadded={false}>
      Wolf whatever craft beer, readymade blue bottle raclette squid next level af franzen heirloom
      90's XOXO beard. Twee schlitz health goth cray. Biodiesel butcher coloring book venmo
      snackwave vexillologist, neutra live-edge enamel pin chia. Roof party church-key fanny pack
      sartorial, flannel shabby chic four loko literally tumeric stumptown coloring book. Organic
      scenester mumblecore cliche, vice readymade helvetica. Glossier 8-bit disrupt affogato pop-up
      post-ironic, la croix cardigan. Gentrify tousled scenester, iceland dreamcatcher master
      cleanse kombucha hoodie mlkshk cardigan seitan before they sold out drinking vinegar ethical
      kickstarter.
    </Paper>
  ))
  .add('zDepth 2', () => (
    <Paper zDepth={2}>
      Wolf whatever craft beer, readymade blue bottle raclette squid next level af franzen heirloom
      90's XOXO beard. Twee schlitz health goth cray. Biodiesel butcher coloring book venmo
      snackwave vexillologist, neutra live-edge enamel pin chia. Roof party church-key fanny pack
      sartorial, flannel shabby chic four loko literally tumeric stumptown coloring book. Organic
      scenester mumblecore cliche, vice readymade helvetica. Glossier 8-bit disrupt affogato pop-up
      post-ironic, la croix cardigan. Gentrify tousled scenester, iceland dreamcatcher master
      cleanse kombucha hoodie mlkshk cardigan seitan before they sold out drinking vinegar ethical
      kickstarter.
    </Paper>
  ));
