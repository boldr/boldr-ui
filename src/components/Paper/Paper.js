/* @flow */
import React, { PureComponent } from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import type { ReactChildren } from '../../types/react.js.flow';

type Props = {
  //  The component to render the paper as.
  component: Function | string,
  //  An optional css className to apply.
  className: ?string,
  // The depth of the paper. This should be a number between 0 - 5. If
  // the depth is 0, it will raise to a depth of 3 on hover.
  zDepth: number,
  // Any children to display in the paper.
  children: ReactChildren,
  //  Boolean if the paper should raise to the `zDepth` of `3` on hover when the initial
  raiseOnHover: boolean,
};

class Paper extends PureComponent {
  static defaultProps = {
    zDepth: 1,
    component: 'div',
  };
  props: Props;
  render() {
    const { component: Component, zDepth, className, raiseOnHover, ...props } = this.props;

    return (
      <Component
        { ...props }
        className={ cn(
          `boldrui-paper boldrui-paper__${zDepth}`,
          {
            'boldrui-paper__0-hover': zDepth === 0 && raiseOnHover,
          },
          className,
        ) }
      />
    );
  }
}

export default Paper;
