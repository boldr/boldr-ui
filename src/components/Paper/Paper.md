# Paper

The `Paper` component is a simple wrappper that adds box-shadow.

You can also use the scss mixin instead of paper.

```scss
  @include md-shadow(5);
```

```javascript
  type Props = {
    //  The component to render the paper as.
    component: Function | string,
    //  An optional css className to apply.
    className: ?string,
    // The depth of the paper. This should be a number between 0 - 5. If
    // the depth is 0, it will raise to a depth of 3 on hover.
    zDepth: number,
    // Any children to display in the paper.
    children: ReactChildren,
    //  Boolean if the paper should raise to the `zDepth` of `3` on hover when the initial
    raiseOnHover: boolean,
  };

  <Paper zDepth={ 2 } raiseOnHover>
    child elements
  </Paper>
```
