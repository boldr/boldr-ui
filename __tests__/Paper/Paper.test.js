/* eslint-env jest */
import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import Paper from '../../src/Paper';

test('<Paper /> matches the snapshot', () => {
  const wrapper = shallow(<Paper zDepth={3} />);

  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
