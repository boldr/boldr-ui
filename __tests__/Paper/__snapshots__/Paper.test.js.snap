// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paper /> matches the snapshot 1`] = `
<div
  className="boldrui-paper boldrui-paper__3"
/>
`;
